File: mobile/types/availability.ts
import { Service } from "./service";

export type AvailabilitySlot = {
  start_time: string;
  end_time: string;
  available: boolean;
};

export type AvailabilityResponse = {
  date: string;
  service_id: number;
  service_name: string;
  duration_minutes: number;
  slots: AvailabilitySlot[];
};

File: backend/app/schemas/availability.py
from datetime import date, time

from pydantic import BaseModel


class AvailabilitySlot(BaseModel):
    start_time: time
    end_time: time
    available: bool


class AvailabilityResponse(BaseModel):
    date: date
    service_id: int
    service_name: str
    duration_minutes: int
    slots: list[AvailabilitySlot]